android/app/src/main/java/org/mozilla/magnet/database/HistoryDatabaseSQL.java

```diff
@@ -0,0 +1,158 @@
+package org.mozilla.magnet.database;
+
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteOpenHelper;
+import android.util.Log;
+
+import org.mozilla.magnet.database.Schema.HistoryEntry;
+
+import java.util.Date;
+
+import static android.provider.BaseColumns._ID;
+import static org.mozilla.magnet.database.Schema.HistoryEntry.COLUMN_NAME_TIME_LAST_SEEN;
+import static org.mozilla.magnet.database.Schema.HistoryEntry.COLUMN_NAME_URL;
+import static org.mozilla.magnet.database.Schema.HistoryEntry.COLUMN_NAME_TIME_FIRST_SEEN;
+
+public class HistoryDatabaseSQL extends SQLiteOpenHelper implements HistoryDatabase {
+    private static final String TAG = History.class.getName();
+
+    /**
+     * Create a new `History` interface.
+     *
+     * @param context
+     */
+    public HistoryDatabaseSQL(Context context) {
+        super(context, Schema.DATABASE_NAME, null, Schema.DATABASE_VERSION);
+    }
+
+    /**
+     * Called once when the database is first created
+     * and each time the database upgrades.
+     *
+     * @param db
+     */
+    @Override
+    public void onCreate(SQLiteDatabase db) {
+        String CREATE_CONTACTS_TABLE = "CREATE TABLE " + HistoryEntry.TABLE_NAME + "("
+                + HistoryEntry._ID + " INTEGER PRIMARY KEY AUTOINCREMENT,"
+                + COLUMN_NAME_URL + " TEXT,"
+                + HistoryEntry.COLUMN_NAME_TIME_FIRST_SEEN + " INT,"
+                + COLUMN_NAME_TIME_LAST_SEEN + " INT"
+                + ")";
+
+        db.execSQL(CREATE_CONTACTS_TABLE);
+    }
+
+    /**
+     * Called when the database version is incremented
+     * giving us the opportunity to migrate tables to
+     * a new schema.
+     *
+     * Right now we're simply dropping the old table
+     * and creating a new replacement.
+     *
+     * @param db
+     * @param oldVersion
+     * @param newVersion
+     */
+    @Override
+    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+        db.execSQL("DROP TABLE IF EXISTS " + HistoryEntry.TABLE_NAME);
+        onCreate(db);
+    }
+
+    /**
+     * Insert a new record.
+     *
+     * @param url
+     */
+    public void insert(String url) {
+        SQLiteDatabase db = this.getWritableDatabase();
+        long now = System.currentTimeMillis() / 1000;
+
+        ContentValues values = new ContentValues();
+        values.put(COLUMN_NAME_URL, url);
+        values.put(HistoryEntry.COLUMN_NAME_TIME_FIRST_SEEN, now);
+        values.put(COLUMN_NAME_TIME_LAST_SEEN, now);
+
+        db.insert(HistoryEntry.TABLE_NAME, null, values);
+        db.close();
```

Comment:
Interesting point. I'm really not sure. If we did keep the reference open then I think we would need to re-architect slightly as when we have more than one table, they would need to share this reference. I'm guessing we'd also need to `.close()` the connection when the app is destroyed?